### PR TITLE
Adding back contributing.md and adjusting for new branch names

### PR DESCRIPTION
**Pull request by brentschmaltz**

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Branch Structure
+* **master**: The latest official GA version, which is currently 5.x
+* **dev**: The dev working branch of master, which is 5.x
+* **dev4x**: The dev working branch of master for 4.x
```
Comment (polita) [inline Contributing.md]: dev4x isn't really a branch of master; it's really a branch of release4x, right?

**Comment by polita** [comment]
One small question, then :shipit:

File: Contributing.md (hunk 1 above)
Comment (brentschmaltz) [inline Contributing.md]: dev4x is the working branch of release4x. We choose that over master4x.

Comment (polita) [inline Contributing.md]: Should this text say 
**dev4x**: The dev working branch for 4.x
So we don't confuse people into thinking it's related to the master branch?

**Pull request merged**